cobbler/items/item.py

```diff
@@ -178,36 +178,70 @@ def __eq__(self, other):
             return self._uid == other.uid
         return False
 
-    def _check_parent_none(self, attribute_name: str, default_value: Any) -> Any:
-        """
-        This method generalizes getting the value from the parent and returning a default in case the parent is not
-        available. In error cases we log this to the log so in case we have a later error this is a starting point.
-
-        :param attribute_name: The name of the attribute to get.
-        :param default_value: The default value which should be returned in case the parent is not available or the
-                              parent doesn't have the required attribute.
-        :raises AttributeError: In case the ``attribute_name`` is not existing on the object.
-        :return: The default value or the value of the parent. None in case the attribute is existing but is not
-                 ``<<inherit>>``.
-        """
-        real_attribute = "_" + attribute_name
-        if hasattr(self, real_attribute):
-            if getattr(self, real_attribute) == enums.VALUE_INHERITED:
-                parent = self.parent
-                if parent is None:
-                    self.logger.info("%s \"%s\" did not have a valid parent but \"%s\" is set to \"<<inherit>>\".",
-                                     type(self), self.name, attribute_name)
-                    return default_value
-                if not hasattr(parent, attribute_name):
-                    self.logger.info("%s \"%s\" did not have a valid parent but \"%s\" is set to \"<<inherit>>\".",
-                                     type(self), self.name, attribute_name)
-                    return default_value
-                return getattr(parent, attribute_name)
+    def _resolve(self, property_name: str) -> Any:
+        """
+        Resolve the ``property_name`` value in the object tree. This function
+        traverses the tree from the object to it's topmost parent and returns
+        the first value that is not inherited. If the the tree does not contain
+        a value the Settings are consulted.
+
+        :param property_name: The property name to resolve.
+        :raises AttributeError: In case one of the objects try to inherit from
+                                a parent that does not have ``property_name``.
+        :return: The resolved value.
+        """
+        attribute = "_" + property_name
+
+        if not hasattr(self, attribute):
+            raise AttributeError("%s \"%s\" does not have property \"%s\""
+                                 % (type(self), self.name, property_name))
+
+        attribute_value = getattr(self, attribute)
+        settings = self.api.settings()
+
+        if attribute_value == enums.VALUE_INHERITED:
+            if self.parent is not None and hasattr(self.parent, property_name):
+                return getattr(self.parent, property_name)
+            elif hasattr(settings, property_name):
+                return getattr(settings, property_name)
```

Comment:
Well I like that but that means that we have to change some settings. For example with `virt_ram` we currently have that in the settings as `default_virt_ram` but we can make that work either through migrations or another if/else branch. So this is not a blocker for me. The error message below is clearly saying what the problem would be...